tests/models/graph_tests.py

```diff
@@ -1481,3 +1271,1001 @@ def test_add_resource_instance_lifecycle(self):
+            label="Widget name",
+        )
+
+        editable_future_graph.widgets[card_x_node_x_widget.pk] = card_x_node_x_widget
+
+        editable_future_graph.save()
+
+        updated_source_graph = updated_source_graph.update_from_editable_future_graph(
+            editable_future_graph=editable_future_graph
+        )
+        editable_future_graph = updated_source_graph.create_editable_future_graph()
+
+        nodegroup_count_before = models.NodeGroup.objects.count()
+        node_count_before = models.Node.objects.count()
+        edge_count_before = models.Edge.objects.count()
+        card_count_before = models.CardModel.objects.count()
+        card_x_node_x_widget_count_before = models.CardXNodeXWidget.objects.count()
+
+        widget = [widget for widget in editable_future_graph.widgets.values()][0]
+        widget.label = "UPDATED_WIDGET_NAME"
+        widget.save()
+
+        updated_source_graph = updated_source_graph.update_from_editable_future_graph(
+            editable_future_graph=editable_future_graph
+        )
+        editable_future_graph = updated_source_graph.create_editable_future_graph()
+
+        serialized_editable_future_graph = JSONDeserializer().deserialize(
+            JSONSerializer().serialize(editable_future_graph)
+        )
+        serialized_updated_source_graph = JSONDeserializer().deserialize(
+            JSONSerializer().serialize(updated_source_graph)
+        )
+
+        self._compare_serialized_updated_source_graph_and_serialized_editable_future_graph(
+            serialized_updated_source_graph, serialized_editable_future_graph
+        )
+
+        updated_widget = [widget for widget in editable_future_graph.widgets.values()][
+            0
+        ]
+        self.assertEqual(updated_widget.label.value, '{"en": "UPDATED_WIDGET_NAME"}')
+
+        nodegroup_count_after = models.NodeGroup.objects.count()
+        node_count_after = models.Node.objects.count()
+        edge_count_after = models.Edge.objects.count()
+        card_count_after = models.CardModel.objects.count()
+        card_x_node_x_widget_count_after = models.CardXNodeXWidget.objects.count()
+
+        self.assertEqual(nodegroup_count_before, nodegroup_count_after)
+        self.assertEqual(node_count_before, node_count_after)
+        self.assertEqual(edge_count_before, edge_count_after)
+        self.assertEqual(card_count_before, card_count_after)
+        self.assertEqual(
+            card_x_node_x_widget_count_before, card_x_node_x_widget_count_after
+        )
+
+    def test_update_from_editable_future_graph_does_not_affect_resources(self):
+        source_graph = Graph.new(name="TEST RESOURCE", is_resource=True, author="TEST")
+        source_graph.save()
+        editable_future_graph = source_graph.create_editable_future_graph()
+
+        nodegroup = models.NodeGroup.objects.create()
+        string_node = models.Node.objects.create(
+            graph=source_graph,
```

Comment:
Let's ensure there's a collector/root node:
```suggestion
            pk=nodegroup.pk,
            graph=source_graph,
```